### ISSUE TITLE
finetuning googlenet ..help

### ISSUE THREAD

**Issue by alevillard** [vlfeat/matconvnet#497]
hi, I'm trying to finetuning GoogLeNet, using the pretrained imagenet-googlenet-dag.mat. Anyone have tried this without errors? 
 now, I see that in this pretrained there are the 3 loss that googlenet have. one question is, should I leave this losses or canceled and leave only the last one? Second question: I have this error in accumulate gradients about matrix dimension:
ans = 

             name: 'objective_cls1'
           inputs: {'prob_cls1'  'label_cls1'}
          outputs: {'objective_cls1'}
           params: {}
     inputIndexes: [62 63]
    outputIndexes: 64
     paramIndexes: [1x0 double]
      forwardTime: []
     backwardTime: []
            block: [1x1 dagnn.Loss]

ans =

           1           1        1000          20

ans = 

             name: 'objective_cls2'
           inputs: {'prob_cls2'  'label_cls2'}
          outputs: {'objective_cls2'}
           params: {}
     inputIndexes: [114 115]
    outputIndexes: 116
     paramIndexes: [1x0 double]
      forwardTime: []
     backwardTime: []
            block: [1x1 dagnn.Loss]

ans =

           1           1        1000          20

ans = 

             name: 'objective'
           inputs: {'prob'  'label'}
          outputs: {'objective'}
           params: {}
     inputIndexes: [163 164]
    outputIndexes: 165
     paramIndexes: [1x0 double]
      forwardTime: []
     backwardTime: []
            block: [1x1 dagnn.Loss]

ans =

     1     1   263    20

ans = 

             name: 'error'
           inputs: {'prob'  'label'}
          outputs: {'error'}
           params: {}
     inputIndexes: [163 164]
    outputIndexes: 166
     paramIndexes: [1x0 double]
      forwardTime: []
     backwardTime: []
            block: [1x1 dagnn.Loss]

ans =

     1     1   263    20

Error using  - 
Matrix dimensions must agree.

Error in cnn_train_dag>accumulate_gradients (line 269)
      state.momentum{p} = opts.momentum * state.momentum{p} ...

Error in cnn_train_dag>process_epoch (line 210)
    state = accumulate_gradients(state, net, opts, batchSize, mmap) ;

Error in cnn_train_dag (line 89)
    stats.train(epoch) = process_epoch(net, state, opts, 'train') ;

Error in googLeNet_train (line 264)
    info = cnn_train_dag(net, imdb, @(i,b) getBatchDisk(bopts,i,b), opts.train, 'val', find(imdb.images.set == 2)) ;

I don't know why the dimension of the other two losses remain at 1000 instead of 263 as I have write in my code for all the losses. Doing a debug of this error in cnn_train_dag about matrix dimension I find this:
-------Debug code from cnn_train_dag
case 'gradient'
      thisDecay = opts.weightDecay * net.params(p).weightDecay; 
      thisLR = state.learningRate * net.params(p).learningRate ;
      p
      OPT=size(opts.momentum)
      STMOM=size(state.momentum{p})
          DECAY=size(thisDecay) 
          PARAMSVAL=size(net.params(p).value)
       PARAMSDER= size(net.params(p).der) 

      state.momentum{p} = opts.momentum * state.momentum{p} ...
        - thisDecay * net.params(p).value ...
        - (1 / batchSize) * net.params(p).der ;
      net.params(p).value = net.params(p).value + thisLR * state.momentum{p} ;

---matlab error:  (note P=42 is a relu, P=43 is the first classifier in googlenet, it have an error in the derivative because I see this size is 0 0 instead of 1 1 512 128 as it should be)

p =

    42

OPT =

     1     1

STMOM =

     1     1

DECAY =

     1     1

PARAMSVAL =

    64     1

PARAMSDER =

    64     1

p =

    43

OPT =

     1     1

STMOM =

     1     1

DECAY =

     1     1

PARAMSVAL =

     1     1   512   128

PARAMSDER =

     0     0

Error using  - 
Matrix dimensions must agree.

Error in cnn_train_dag>accumulate_gradients (line 269)
      state.momentum{p} = opts.momentum * state.momentum{p} ...

**Comment by maxiaoyuelun** [comment]
@alevillard I've met the same problem like you, could you tell me how to solve it?

**Comment by alevillard** [comment]
I should delete the first two losses from the network and leave only the last classifier. It should works fine for finetuning. 
The first two losses were useful for the google team to make the training, as googlenet paper say about that.
